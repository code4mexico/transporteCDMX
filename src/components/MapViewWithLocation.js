/* eslint-disable react/jsx-props-no-spreading */
import React, { useState, useRef } from 'react'
import PropTypes from 'prop-types'
import MapView from 'react-native-maps'
import { coordinateToRegion } from '../utils/location'

const MapViewWithLocation = props => {
  const [followsUserLocation, setFollowsUserLocation] = useState(true)
  const mapRef = useRef('map')
  const { children, ...mapProps } = props

  const _onUserLocationChange = event => {
    if (mapRef?.current && event && followsUserLocation) {
      mapRef.current.animateToRegion(coordinateToRegion(event.nativeEvent.coordinate))
      setFollowsUserLocation(false)
    }
  }

  return (
    <MapView
      ref={mapRef}
      onUserLocationChange={_onUserLocationChange}
      followsUserLocation={followsUserLocation}
      showsUserLocation
      {...mapProps}>
      {children}
    </MapView>
  )
}

MapViewWithLocation.propTypes = {
  children: PropTypes.node,
}

MapViewWithLocation.defaultProps = {
  children: [],
}

export default MapViewWithLocation
