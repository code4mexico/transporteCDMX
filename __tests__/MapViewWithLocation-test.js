import React from 'react'
import { View } from 'react-native'
import { shallow } from 'enzyme'
import MapViewWithLocation from '../src/components/MapViewWithLocation'

describe('MapViewWithLocation', () => {
  it('it renders correctly', () => {
    const wrapper = shallow(<MapViewWithLocation />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('MapViewWithLocation', () => {
  it('it renders correctly with Children', () => {
    const wrapper = shallow(
      <MapViewWithLocation>
        <View />
      </MapViewWithLocation>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
