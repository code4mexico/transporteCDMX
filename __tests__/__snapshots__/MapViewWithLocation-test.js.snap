// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapViewWithLocation it renders correctly 1`] = `
<MapView
  followsUserLocation={true}
  onUserLocationChange={[Function]}
  showsUserLocation={true}
/>
`;

exports[`MapViewWithLocation it renders correctly with Children 1`] = `
<MapView
  followsUserLocation={true}
  onUserLocationChange={[Function]}
  showsUserLocation={true}
>
  <View />
</MapView>
`;
